README.md

```diff
@@ -23,121 +23,142 @@ Sourcebot is a fast code indexing and search tool for your codebases. It is buil
+Sourcebot needs a config file to tell it which repositories to index. By default, Sourcebot will look for a file called `config.json` within the mounted workspace.
 
-    </details>
+Create a new file called `config.json` inside your workspace and paste in the following sample config which loads the `sourcebot` github repo:
+```
+{
+    "$schema": "https://raw.githubusercontent.com/TaqlaAI/sourcebot/main/schemas/index.json",
+    "Configs": [
+        {
+            "Type": "github",
+            "GitHubOrg": "TaqlaAI",
+            "Name": "sourcebot"
+        }
+    ]
+}
+```
 
-    <details>
-    <summary><img src="https://bitbucket.org/favicon.ico" width="16" height="16" /> BitBucket</summary>
+##### Changing the config file
+The config file follows the schema defined [here](https://raw.githubusercontent.com/TaqlaAI/sourcebot/main/schemas/index.json). You can define multiple repos by adding config objects into the `Configs` list:
+
+<pre>
+{
+    "$schema": "https://raw.githubusercontent.com/TaqlaAI/sourcebot/main/schemas/index.json",
+    "Configs": [
+        {
+            "Type": "github",
+            "GitHubOrg": "TaqlaAI", <b>// github orgs must define the GithubOrg field</b>
+            "Name": "sourcebot"     <b>// name of the github repository to index (regex match)</b>
+        },
+        {
+            "Type": "github",
+            "GithubUser": "ssloy",  <b>// if indexing a github repo owned by a user, set the GithubUser field</b>
+            "Name": "tinyrenderer"  <b>// name of the github repository to index (regex match)</b>
+        }
+    ] 
+}
+</pre>
+
+TODO: add examples for GitLab and BitBucket above
+
+### Mount your workspace and run Sourcebot
+We can now run Sourcebot, but this time we mount the workspace folder so that it can pickup our custom config: 
+
+> [!NOTE]
+> This command assumes that you're running it from within your Sourcebot workspace. If you're not, replace `$(pwd)` below with the absolute path of your Sourcebot workspace
 
-    TODO
+```
+docker run -p 3000:3000 --rm --name sourcebot -v $(pwd):/data ghcr.io/taqlaai/sourcebot:main
+```
 
-    </details>
-    </div>
+Navigate to `localhost:3000` in your browser to use Sourcebot to search the repositories you've defined in your config!
 
-3. Launch the latest image from the [ghcr registry](https://github.com/TaqlaAI/sourcebot/pkgs/container/sourcebot):
+### (Optional) Provide an access token to index private repositories
+In order to allow Sourcebot to index your private repositories, you must provide it with an access token.
 
-    <div>
-    <details open>
-    <summary>
-    <picture>
-        <source media="(prefers-color-scheme: dark)" srcset=".github/images/github-favicon-inverted.png">
-        <img src="https://github.com/favicon.ico" width="16" height="16" alt="GitHub icon">
-    </picture>
-    GitHub
-    </summary>
+<div>
+<details>
+<summary><img src="https://github.com/favicon.ico" width="16" height="16" /> GitHub</summary>
```

Comment:
Nit: the github logo here is not visible when in darkmode:
<img width="855" alt="image" src="https://github.com/user-attachments/assets/400f2c9f-67a7-401f-ba28-04566c1a7273">

If you take a look at other instances of the logo, there are examples of how to handle dark mode